library/src/main/java/com/datatorrent/lib/state/BucketedState.java

```diff
@@ -29,11 +31,17 @@
 public interface BucketedState
 {
   /**
+   * An expired value. In some implementations where bucketId is time related then the event can be old and
+   * the get methods- getSync & getAsync return this special slice instance.
+   */
+  Slice EXPIRED = new Slice(Ints.toByteArray(-1));
```

Comment:
Couldn't -1 be a valid user value ?
Might be better to use invalid slices, e.g. Slice(null, -1, -1). Or, if you want to be more user friendly, extend Slice and add an isExpired() method which checks offset and length against -1.